StarRailBattleSim/src/battleLogic/log/lines/metrics/PostCombatPlayerMetrics.java

```diff
@@ -0,0 +1,67 @@
+import battleLogic.log.Logger;
+import characters.AbstractCharacter;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PostCombatPlayerMetrics implements Loggable {
+
+    public final AbstractCharacter player;
+    public final Map<StatType, Float> finalStats = new HashMap<>();
+    public final Map<String, Boolean> relicSets = new HashMap<>();
+    public final int turnsTaken;
+    public final int basics;
+    public final int skills;
+    public final int ultimates;
+    public final List<AbstractCharacter.MoveType> rotation;
+    public final Map<String, String> characterSpecificMetrics;
+
+    public PostCombatPlayerMetrics(AbstractCharacter player) {
+        this.player = player;
+
+        finalStats.put(StatType.ATK, player.getFinalAttack());
+        finalStats.put(StatType.DEF, player.getFinalDefense());
+        finalStats.put(StatType.HP, player.getFinalHP());
+        finalStats.put(StatType.SPD, player.getFinalSpeed());
+        finalStats.put(StatType.DMG, player.getTotalSameElementDamageBonus());
+        finalStats.put(StatType.CRIT, player.getTotalCritChance());
+        finalStats.put(StatType.CRITDMG, player.getTotalCritDamage());
+        finalStats.put(StatType.BREAK, player.getTotalBreakEffect());
+        finalStats.put(StatType.EHR, player.getTotalEHR());
+        finalStats.put(StatType.ERR, player.getTotalERR());
+        finalStats.put(StatType.RES, player.getTotalEffectRes());
+        player.relicSetBonus.forEach(r -> relicSets.put(r.toString(), r.isFullSet()));
+
+        this.turnsTaken = player.numTurnsMetric;
+        this.basics = player.numBasicsMetric;
+        this.skills = player.numSkillsMetric;
+        this.ultimates = player.numUltsMetric;
+        this.rotation = player.moveHistory;
+        this.characterSpecificMetrics = new HashMap<>(player.getCharacterSpecificMetricMap());
+    }
+
+    @Override
+    public String asString() {
+        String statsString = "";
+        String gearString = String.format("Metrics for %s \nLightcone: %s \nRelic Set Bonuses: ", player.name, player.lightcone);
+        gearString += player.relicSetBonus;
+        statsString = gearString + String.format("\nAfter combat stats \nAtk: %.3f \nDef: %.3f \nHP: %.3f \nSpeed: %.3f \nSame Element Damage Bonus: %.3f \nCrit Chance: %.3f%% \nCrit Damage: %.3f%% \nBreak Effect: %.3f%%",
+                finalStats.get(StatType.ATK), finalStats.get(StatType.DEF), finalStats.get(StatType.HP), finalStats.get(StatType.SPD), finalStats.get(StatType.DMG), finalStats.get(StatType.CRIT), finalStats.get(StatType.CRITDMG), finalStats.get(StatType.BREAK));
+
+        StringBuilder metrics = new StringBuilder(statsString + String.format("\nCombat Metrics \nTurns taken: %d \nBasics: %d \nSkills: %d \nUltimates: %d \nRotation: %s",
+               turnsTaken, basics, skills, ultimates, rotation));
+        for (Map.Entry<String, String> entry : this.characterSpecificMetrics.entrySet()) {
```

Comment:
I noticed these metrics got printed in a wacky order, the way I had them before is that I used an arraylist to determine what order to print these metrics in
```
for (String metric : getOrderedCharacterSpecificMetricsKeys()) {
            metrics.append("\n").append(metric).append(": ").append(metricsMap.get(metric));
        }
```